sconstruct

```diff
@@ -545,3 +545,65 @@ env.Alias('appx',[installed_appx_storeSubmission,installed_appx_sideLoadable])
 env.Default(dist)
 
 env.SConscript("tests/sconscript", exports=["env", "sourceDir", "pot"])
+
+
+def generate_module_list(target, source, env):
+	""" 
+	Generate a list of Python modules from compiled '.pyc' files within a specified zipfile.
+
+    This function lists all '.pyc' files contained in a given zipfile, converts their paths to Python module
+    format, performs a sanity check to ensure a specific module is present, and then writes the list of
+    modules to a specified output file for packaging as an AppVeyor artefact.
+
+    Args:
+        target (list of SCons.Node.FS.File): A single element list containing the target file node where the 
+											 list of modules will be written.
+        source (list of SCons.Node.FS.File): A single element list containing the source file node to be 
+											 processed. It should contain only the zipfile from which the 
+											 '.pyc' files will be listed.
+        env (SCons.Environment): The SCons environment context under which this function is executed. This
+                                 parameter provides access to SCons construction variables, methods, and tools.
+
+    Returns:
+        None: The function does not return anything but writes the list of modules to the target file.
+
+    Raises:
+        ValueError: If the specified sanity check module (e.g., 'NVDAObjects.UIA') is not found in the
+					list of modules extracted from the zipfile.
+        FileNotFoundError: If the specified zipfile in `source` does not exist.
+	"""
+	import zipfile
+	import re
+
+	# List all .pyc files in the library.zip
+	library_zip_path = str(source[0])
+	if not os.path.exists(library_zip_path):
+		raise FileNotFoundError(f"The zipfile {library_zip_path} does not exist.")
+	pyc_files = [f for f in zipfile.ZipFile(library_zip_path, 'r').namelist() if f.endswith('.pyc')]
+
+	# Convert the file paths to python module format
+	# eg: NVDAObjects/IAccessible/__init__.pyc --> NVDAObjects.IAccessible
+	imported_modules = sorted(set([
+		re.sub(r'(.__init__|.__version__|._version)?\.pyc$', '', module_path).replace('/', '.')
+		for module_path in pyc_files
+	]))
+
+	# Sanity check for something guaranteed to be in library.zip
+	if "NVDAObjects.UIA" not in imported_modules:
+		raise ValueError(f"Expected module NVDAObjects.UIA not found in the zipfile.")
+
+	# Store the list to file
+	with open(str(target[0]), 'w') as file:

```

Comment:
This will open the file using whatever encoding is the default ANSI code page on a given system, meaning it may behave differently depending on the Windows language used by a particular developer. This was causing some issues in the past, so its safer to be explicit here:
```suggestion
	with open(str(target[0]), 'w', encoding="utf-8") as file:
```